test/integration/retryable-writes/retryable_writes.spec.prose.test.ts

```diff
@@ -299,5 +296,48 @@ describe('Retryable Writes Spec Prose', () => {
         expect(insertResult).to.have.property('code', 91);
       }
     );
+
+    // This is an extra test that is a complimentary test to prose test #3. We basically want to
+    // test that in the case of a write concern error with ok: 1 in the response, that
+    // a command succeeded event is emitted but that the driver still treats it as a failure
+    // and retries. So for the success, we check the error code if exists, and since the retry
+    // must succeed, we fail if any command failed event occurs on insert.
+    it(
+      'emits a command succeeded event for write concern errors with ok: 1',
+      { requires: { topology: 'replicaset', mongodb: '>=4.2.9' } },
+      async () => {
+        // Generate a write concern error to assert that we get a command
+        // suceeded event but the operation will retry because it was an
+        // actual write concern error.
+        await client.db('admin').command({
+          configureFailPoint: 'failCommand',
+          mode: { times: 1 },
+          data: {
+            writeConcernError: {
+              code: 91,
+              errorLabels: ['RetryableWriteError']
+            },
+            failCommands: ['insert']
+          }
+        });
+
+        const willBeCommandSucceeded = once(client, 'commandSucceeded').catch(error => error);
+        const willBeCommandFailed = once(client, 'commandFailed', {
+          signal: AbortSignal.timeout(1000)
+        }).catch(error => error);
```

Comment:
I looked at the docs before suggesting this and it said AbortSignal was available in 14.17+ and we're testing on 14.21 so not sure why it's erroring 😕 


Stealing this from our unified runner: https://github.com/mongodb/node-mongodb-native/blob/main/test/tools/unified-spec-runner/operations.ts#L464-L476
```typescript
const willBeCommandFailed = Promise.race([
  once(client, 'commandFailed'), 
  sleep(1000).then(() => Promise.reject(new Error('timeout'))
]).catch(error => error);
```

Reply:
Updated and rebased.